### ISSUE TITLE
--fail-fast silently passes when offenses are found

### ISSUE THREAD

**Issue by JorgeLeites** [rubocop/rubocop#15318]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False negative

### Steps to reproduce the problem

Steps to reproduce

1. Create a file with a RuboCop offense
2. Run with --fail-fast:

```
bundle exec rubocop -F path/to/file
```

### Expected behavior

Expected behavior

- Offenses in the file are reported.
- The file is counted as inspected.
- Exit code is non-zero (same as without -F).
With -F on a full run, inspection stops after the first file with offenses, but that file’s offenses are still shown.

### Actual behavior

With -F:

```
Inspecting 1 file


0 files inspected, no offenses detected
```
- Exit code is 0 (success).
- No offenses are printed.
- On a full project run, recently modified files are checked first (mtime order). If the first file with offenses triggers fail-fast, CI/local lint can pass even when bundle exec rubocop (without -F) reports many offenses.
Example from a real project:

```
$ bundle exec rubocop
2960 files inspected, 22 offenses detected, 20 offenses autocorrectable
$ bundle exec rubocop -F
6 files inspected, no offenses detected
```

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
$ bundle exec rubocop -V
1.88.0 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 3.2, running on ruby 3.3.11) [arm64-darwin25]
  - rubocop-rails 2.35.4
  - rubocop-performance 1.26.1
```
